compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -762,9 +762,9 @@ class SpaceEngine(using Context) extends SpaceLogic {
         if (ctx.definitions.isTupleType(tp))
           params(tp).map(_ => "_").mkString("(", ", ", ")")
         else if (scalaListType.isRef(sym))
-          if (flattenList) "_: _*" else "_: List"
+          if (flattenList) "_*" else "_: List"
         else if (scalaConsType.isRef(sym))
```

Comment:
Isn't '_*' parsed as a single token? Should it be '_ *' instead?

Reply:
From the syntax specification, `_*` will be parsed correctly, and the implementation agrees with the syntax: https://scastie.scala-lang.org/EvlND8SaTMGnJjyhaqdRIA

Reply:
This snippet crashes the compiler.

Reply:
I enabled `-Xprint:typer` in the above, the output is noisy. Here is a clean one: https://scastie.scala-lang.org/KnwXPu1kTQKYp9EGI8qrQw